app/pods/publications/publication/publication-activities/index/controller.js

```diff
@@ -0,0 +1,118 @@
+import Controller from '@ember/controller';
+import { inject as service } from '@ember/service';
+import { action } from '@ember/object';
+import { tracked } from '@glimmer/tracking';
+import { dropTask, task } from 'ember-concurrency';
+/* eslint-disable no-unused-vars */
+import PublicationActivity from '../../../../../models/publication-activity';
+import DecisionModel from '../../../../../models/decision';
+/* eslint-enable no-unused-vars */
+
+/** @param {PublicationActivity[]} publicationActivities */
+export async function createTimeline(publicationActivities) {
+  let publicationTimeline = [];
+  for (const publicationActivity of publicationActivities) {
+    const publicationRequestEvent = await PublicationRequestEvent.create(
+      publicationActivity
+    );
+    publicationTimeline.push(publicationRequestEvent);
+    let decisions = await publicationActivity.decisions;
+    decisions = decisions.toArray();
+    for (const decision of decisions) {
+      const publicationEvent = await PublicationEvent.create(decision);
+      publicationTimeline.push(publicationEvent);
+    }
+  }
+  publicationTimeline = publicationTimeline.sortBy('date').reverse();
+  return publicationTimeline;
+}
+
+export class PublicationRequestEvent {
```

Comment:
Ik heb hier gekozen voor row-objecten, zoals dat was in de oude proofs-route. Ik heb de "Vertalingen"-pull-request gedeeltelijk doorgenomen. Deze is nog niet af, maar heb gezien dat Tanee hier voor extra components heeft gekozen. Dit kan ik ook doen (Al zou ik de row-objecten behouden om ze op datum te sorteren.)

Reply:
Als ik je intentie goed begrijp is het hier de bedoeling om objecten te maken die dezelfde 'interface', nl. `title` en `date`, aanbieden, om zo in de template makkelijker gewoon over de lijst te kunnen loopen en weergeven?

Ik vind de static `create` methods wat verwarrend. Die lijken voor mij te impliceren dat er speciale dingen moeten gebeuren die niet met een gewone constructor kunnen, maar als je er dan naar gaat kijken, is dat niet het geval (of mis ik iets?).

Kunnen we de 2 aparte classes misschien mergen naar 1 class `PublicationEvent` met een `@tracked model` dat zowel een `publication-activity` als een `decision` kan zijn? Op die class dan `isRequest` en `isPublication` als get-function definieren die op basis van `this.args.model.constructor.modelName` de correcte boolean teruggeeft. En in de get-functions van `title` en `date` dan met `if (this.isRequest) { ... } else if (this.isPublication) { ... }` de juiste waarde teruggeven. Dat maakt voor mij alleszins de intentie van die class iets beter duidelijk.

Reply:
- [x] de `create`-method is weggewerkt.
- [ ] Ik heb de opdeling volgens model-type wel behouden. Het doel van deze klasse is net om om te kunnnen met de verschillen tussen `request-activity` en `publication-activity`.